instrumented gate: observe permitted and non-permitted queries separately

dimitarvdimitrov opened the pull request:
This distinction helps when debugging increased congestion on a gate.


**Checklist**
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

pracucci reviewed (approved):
LGTM! I would just add a CHANGELOG entry. Thanks!

dimitarvdimitrov commented:
> LGTM! I would just add a CHANGELOG entry. Thanks!

thanks for the review. I added further separation of outcomes between context cancelled and deadline exceeded (and now "other"). Also added a changelog entry. I rebase the branch, so you can look at only the last 2 commits if you want partial diff.

dimitarvdimitrov commented:
i also mistakenly committed some other changes; cleared them out in another 2 commits

pracucci reviewed (approved):
LGTM thanks!
